File: desktop/BeautyEstiva.Desktop/BeautyEstiva.Desktop/Models/CustomerModels.cs
using System.Text.Json.Serialization;

namespace BeautyEstiva.Desktop.Models;

public class CustomerListItem
{
    [JsonPropertyName("id")]
    public int Id { get; set; }

    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("surname")]
    public string Surname { get; set; } = string.Empty;

    [JsonPropertyName("fullName")]
    public string FullName { get; set; } = string.Empty;

    [JsonPropertyName("phone")]
    public string Phone { get; set; } = string.Empty;

    [JsonPropertyName("email")]
    public string? Email { get; set; }

    [JsonPropertyName("totalAppointments")]
    public int TotalAppointments { get; set; }

    [JsonPropertyName("lastAppointmentDate")]
    public string? LastAppointmentDate { get; set; }

    [JsonPropertyName("loyaltyPoints")]
    public int LoyaltyPoints { get; set; }

    [JsonPropertyName("totalSpent")]
    public decimal TotalSpent { get; set; }

    [JsonPropertyName("totalVisits")]
    public int TotalVisits { get; set; }

    [JsonPropertyName("segment")]
    public string Segment { get; set; } = string.Empty;

    [JsonPropertyName("tags")]
    public List<string> Tags { get; set; } = [];

    [JsonPropertyName("customerSince")]
    public string? CustomerSince { get; set; }
}

public class CustomerCreate
{
    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("surname")]
    public string Surname { get; set; } = string.Empty;

    [JsonPropertyName("phone")]
    public string? Phone { get; set; }

    [JsonPropertyName("email")]
    public string? Email { get; set; }

    [JsonPropertyName("birthDate")]
    public string? BirthDate { get; set; }

    [JsonPropertyName("notes")]
    public string? Notes { get; set; }

    [JsonPropertyName("allergies")]
    public string? Allergies { get; set; }

    [JsonPropertyName("preferences")]
    public string? Preferences { get; set; }

    [JsonPropertyName("referralSource")]
    public string? ReferralSource { get; set; }

    [JsonPropertyName("preferredStaffId")]
    public int? PreferredStaffId { get; set; }

    [JsonPropertyName("tags")]
    public List<string>? Tags { get; set; }
}

public class CustomerDetail
{
    [JsonPropertyName("id")]
    public int Id { get; set; }

    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("surname")]
    public string Surname { get; set; } = string.Empty;

    [JsonPropertyName("fullName")]
    public string FullName { get; set; } = string.Empty;

    [JsonPropertyName("phone")]
    public string Phone { get; set; } = string.Empty;

    [JsonPropertyName("email")]
    public string? Email { get; set; }

    [JsonPropertyName("birthDate")]
    public string? BirthDate { get; set; }

    [JsonPropertyName("notes")]
    public string? Notes { get; set; }

    [JsonPropertyName("totalAppointments")]
    public int TotalAppointments { get; set; }

    [JsonPropertyName("loyaltyPoints")]
    public int LoyaltyPoints { get; set; }

    [JsonPropertyName("totalSpent")]
    public decimal TotalSpent { get; set; }

    [JsonPropertyName("totalVisits")]
    public int TotalVisits { get; set; }

    [JsonPropertyName("lastVisitDate")]
    public string? LastVisitDate { get; set; }

    [JsonPropertyName("customerSince")]
    public string? CustomerSince { get; set; }

    [JsonPropertyName("preferredStaffName")]
    public string? PreferredStaffName { get; set; }

    [JsonPropertyName("allergies")]
    public string? Allergies { get; set; }

    [JsonPropertyName("preferences")]
    public string? Preferences { get; set; }

    [JsonPropertyName("tags")]
    public List<string> Tags { get; set; } = [];

    [JsonPropertyName("segment")]
    public string Segment { get; set; } = string.Empty;

    [JsonPropertyName("averageSpendPerVisit")]
    public decimal AverageSpendPerVisit { get; set; }
}

File: flutter/lib/core/models/customer_models.dart
class CustomerListItem {
  final int id;
  final String name;
  final String surname;
  final String fullName;
  final String phone;
  final String? email;
  final int totalAppointments;
  final int totalVisits;
  final double totalSpent;
  final int loyaltyPoints;
  final String segment;
  final List<String> tags;

  CustomerListItem({
    required this.id,
    required this.name,
    required this.surname,
    required this.fullName,
    required this.phone,
    this.email,
    required this.totalAppointments,
    required this.totalVisits,
    required this.totalSpent,
    required this.loyaltyPoints,
    required this.segment,
    required this.tags,
  });

  factory CustomerListItem.fromJson(Map<String, dynamic> json) => CustomerListItem(
        id: json['id'] ?? 0,
        name: json['name'] ?? '',
        surname: json['surname'] ?? '',
        fullName: json['fullName'] ?? '',
        phone: json['phone'] ?? '',
        email: json['email'],
        totalAppointments: json['totalAppointments'] ?? 0,
        totalVisits: json['totalVisits'] ?? 0,
        totalSpent: (json['totalSpent'] ?? 0).toDouble(),
        loyaltyPoints: json['loyaltyPoints'] ?? 0,
        segment: json['segment'] ?? '',
        tags: List<String>.from(json['tags'] ?? []),
      );
}

class CustomerCreate {
  final String name;
  final String surname;
  final String? phone;
  final String? email;
  final String? notes;

  CustomerCreate({required this.name, required this.surname, this.phone, this.email, this.notes});

  Map<String, dynamic> toJson() => {
        'name': name,
        'surname': surname,
        if (phone != null) 'phone': phone,
        if (email != null) 'email': email,
        if (notes != null) 'notes': notes,
      };
}
